specs/Workers.md

```diff
@@ -0,0 +1,1053 @@
+  /// Get the service worker manager to monitor service worker registrations and interact with the service worker associated with the current profile.
+  ///
+  /// The changes would apply to the context of the user profile. That is, other WebViews under the same user profile could be affected.
+  ///
+  /// \snippet ScenarioServiceWorkerManager.cpp ServiceWorkerManager
+  [propget] HRESULT ServiceWorkerManager([out, retval] ICoreWebView2ServiceWorkerManager** value);
+
+
+  /// Get the shared worker manager to monitor shared worker creations and interact with the shared worker associated with the current profile.
+  ///
+  /// The changes would apply to the context of the user profile. That is, other WebViews under the same user profile could be affected.
+  ///
+  /// \snippet ScenarioSharedWorkerManager.cpp SharedWorkerManager
+  [propget] HRESULT SharedWorkerManager([out, retval] ICoreWebView2SharedWorkerManager** value);
+}
+
+[uuid(4e07e562-8db7-5815-907b-6d89a253a974), object, pointer_default(unique)]
+interface ICoreWebView2ServiceWorkerManager : IUnknown {
+  /// Adds an event handler for the `ServiceWorkerRegistered` event.
+  ///
+  /// A ServiceWorker is a specific type of worker that takes a JavaScript file that can control the web-page/site that it is associated with,
+  /// intercepting and modifying navigation and resource requests, and caching resources in a very granular fashion to give you complete control
+  /// over how app behaves in certain situations.
+  ///
+  /// Service workers essentially act as proxy servers that sit between web applications, the browser, and the network (when available).
+  /// Unlike Shared Workers, which have their own separate global scope, Service Workers have no DOM access and run in a different context.
+  ///
+  /// This event is raised when a web application registers a service worker using the
+  /// `navigator.serviceWorker.register("/sw.js")` method. See the
+  /// [Service Worker Registration](https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerRegistration)
+  /// for more information.
+  ///
+  ///
+  /// \snippet ScenarioServiceWorkerManager.cpp ServiceWorkerRegistered
+  HRESULT add_ServiceWorkerRegistered(
+      [in] ICoreWebView2ServiceWorkerRegisteredEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Removes an event handler previously added with `add_ServiceWorkerRegistered`.
+  HRESULT remove_ServiceWorkerRegistered(
+      [in] EventRegistrationToken token);
+
+
+  /// Gets a list of the service worker registrations under the same profile.
+  HRESULT GetServiceWorkerRegistrations(
+      [in] ICoreWebView2GetServiceWorkerRegistrationsCompletedHandler* handler
+  );
+
+  /// Gets the service worker registration object associated with the specified scope. If a service worker has been registered
+  /// for the given scope, it gets the corresponding `CoreWebView2ServiceWorkerRegistration` object.
```

Comment:
What is scope in this context? When you register a service worker in script you can optionally specify a scope that is a path. But that's within the context of the service worker's origin right? In this case the manager is for the whole profile so multiple different origins. Does the scope string here include the origin? More documentation is required here to explain what scope is. And also how it gets compared (casing / IDN for hostnames and such). An example will also be nice

Reply:
The scope here includes origin and path which is specified during registration which matches with the [Scope ](https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerRegistration/scope)property in DOM SWregistration . Updated the documentation